core/templates/components/skill-selector/select-skill-modal.component.ts

```diff
@@ -19,20 +19,25 @@ import { Component } from '@angular/core';
 import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
 import { ConfirmOrCancelModal } from 'components/common-layout-directives/common-elements/confirm-or-cancel-modal.component';
 import { ShortSkillSummary } from 'domain/skill/short-skill-summary.model';
-import { SkillSummaryBackendDict } from 'domain/skill/skill-summary.model';
+import { SkillSummary, SkillSummaryBackendDict } from 'domain/skill/skill-summary.model';
 import { SkillsCategorizedByTopics } from 'pages/topics-and-skills-dashboard-page/skills-list/skills-list.component';
 
+type TotalSkills = (
+  (SkillSummaryBackendDict | SkillSummary | ShortSkillSummary)[]);
```

Comment:
Why some many different possible "versions" of skills?

Reply:
Also this file seems to be assigned to someone else (#15049)

Reply:
Actually in this file we are selecting 3 types of skills, example `skillSummaries`, `untriagedSkillSummaries`, `categorizedSkills` and store in array totalSkills whose type is `TotalSkills`.

Reply:
This doesn't really make sense tbh. What, conceptually, is "totalSkills"? 

Even if it's an array of skills or skill summaries or skill objects etc., all items in that list should have the same type. Otherwise, the code might need to be refactored.

Reply:
Done

Reply:
Done for now, but Sean why we have two types `ShortSkillSummaryBackendDict` and `SkillSummaryBackendDict`, actually both are pointed to same??

Reply:
Perhaps -- what is the composition (i.e. the set of key-value pairs) for each of these? If they are the same then I agree it makes sense to use only one of the types, but this needs to be investigated.

Reply:
like `ShortSkillSummaryBackendDict` uses id and description only, of `SkillSummaryBackendDict`. 
Can we make an issue for that ? because it is not related to this PR ? Need proper changes of types in many files.

Reply:
It sounds like they are different, then. So they should be represented by different objects. (Sometimes the abbreviated version is needed if you are fetching a lot of them at once and only want the abbreviated information; this prevents a lot of unnecessary data being downloaded to the user's browser.)

Reply:
My point still applies, this file is assigned to someone else, so I think it should not be done here (#15179).

Reply:
For me this file `select-skill-modal.component.ts` is one of the dependencies, and in #15179 he directly do so. After merge of my PR, he will take a pull, And move ahead.

Reply:
@JeeveshGarg Okay, but please explain it to the author of #15179 so that he knows what is going on.